### PR TITLE
#690 Add indexes to Menas collections

### PR DESCRIPTION
**Pull request by GeorgiChochov**

### CODE REVIEW

**Comment by GeorgiChochov** [comment]
We have preexisting duplicates (simultaneous `Runs` created with the same `dataset` + `datasetVersion` + `runId`) in our production database's `run` collection. Let's park this until that is resolved.

File: menas/src/main/scala/za/co/absa/enceladus/menas/services/ModelService.scala [hunk 1]
```diff
@@ -15,19 +15,23 @@
...
+
+  val log: Logger = LoggerFactory.getLogger(this.getClass)
+
   def isUniqueName(name: String): Future[Boolean] = {
     mongoRepository.isUniqueName(name)
   }
 
-  def create(item: C): Future[Completed] = {
-    mongoRepository.create(item)
+  def create(item: C): Future[C] = {
```
[the remark this answers is not in the document]
Comment (GeorgiChochov) [reply menas/src/main/scala/za/co/absa/enceladus/menas/services/ModelService.scala]: Excellent point! Thanks!

File: menas/src/test/scala/za/co/absa/enceladus/menas/integration/controllers/SchemaApiIntegrationSuite.scala [hunk 2]
```diff
@@ -19,26 +19,105 @@ import org.junit.runner.RunWith
...
+          assertCreated(response)
+
+          val actual = response.getBody
+          val expected = toExpected(schema, actual)
+          assert(actual == expected)
+        }
+      }
+    }
+
+    "return 400" when {
```
Comment (yruslan) [inline menas/src/test/scala/za/co/absa/enceladus/menas/integration/controllers/SchemaApiIntegrationSuite.scala]: Consider maybe switching to 409 (conflict) sometime in the future (not in this PR)?

Comment (GeorgiChochov) [reply menas/src/test/scala/za/co/absa/enceladus/menas/integration/controllers/SchemaApiIntegrationSuite.scala]: Yeah, I thought about it. Probably a good idea!

Comment (GeorgiChochov) [reply menas/src/test/scala/za/co/absa/enceladus/menas/integration/controllers/SchemaApiIntegrationSuite.scala]: Created #738 for this.

**Pull request merged**